### PR TITLE
[linter] Permit examples to contain arguments that start with 'az ' or contain a pipe '|' .

### PR DESCRIPTION
**Pull request by adewaleo**
This only adds nested args as commands if they are interpolation / substitution arguments.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by adewaleo** [comment]
Closing this PR as this issue will be fixed here(https://github.com/Azure/azure-cli-dev-tools/issues/30)

**Comment by adewaleo** [comment]
Closing this PR as this issue will be fixed here(https://github.com/Azure/azure-cli-dev-tools/issues/30)

**Comment by adewaleo** [comment]
@williexu I have added the changes.

File: tools/automation/cli_linter/rules/help_rules.py [hunk 1]
```diff
@@ -116,30 +116,35 @@ def _extract_commands_from_example(example_text):
...
     commands = example_text.splitlines()
     processed_commands = []
     for command in commands:  # filter out commands
         if command.startswith("az"):
             processed_commands.append(command)
-        elif "az " in command:  # some commands start with "$(az ..." and even "`az in one case"
+        elif "az " in command:  # some commands start with "$(az ..." and even "`az in one case" deal with closing brackets later.
             idx = command.find("az ")
             command = command[idx:]
+            command = command.rstrip(")`")
```
Comment (williexu) [inline tools/automation/cli_linter/rules/help_rules.py]: Let's prevent a scenario like:
```
az baz --foo $(az foo) --bar
```
from seeing `az foo) --bar` as a inner command.

We'll ignore more than one level of nestedness per our discussion :)

File: tools/automation/cli_linter/rules/help_rules.py [hunk 2]
```diff
@@ -116,30 +116,35 @@ def _extract_commands_from_example(example_text):
...
             processed_commands.append(command)
 
     return processed_commands
 
 
 def _process_command_args(command_args):
+
+    def _is_special_arg(arg):
+        is_special = (arg.startswith("$(") and arg.endswith(")")) or (arg.startswith("`") and arg.endswith("`"))
```
Comment (williexu) [inline tools/automation/cli_linter/rules/help_rules.py]: Args can also be something like: `az foo --bar $(az baz)=1`. A regex might be a better choice

**Comment by adewaleo** [comment]
@williexu It does, just confirmed again and merged.

**Pull request merged**